dao/src/test/java/com/epam/esm/dao/impl/GiftCertificateDaoTest.java

```diff
@@ -0,0 +1,180 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.config.DaoTestConfig;
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.entity.GiftCertificate;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {DaoTestConfig.class})
+@ActiveProfiles("test")
+public class GiftCertificateDaoTest {
+
+    @Autowired
+    private GiftCertificateDao giftCertificateDao;
+
+    private static GiftCertificate certificate1;
+    private static GiftCertificate certificate2;
+    private static GiftCertificate certificate3;
+
+    @BeforeAll
+    static void init() {
+        certificate1 = new GiftCertificate(1L, "certificate1", "descripiton1",
+                new BigDecimal("1.5"), 10, "2021-03-29T11:24:04.643", "2021-03-29T11:30:18.653");
+        certificate2 = new GiftCertificate(2L, "certificate2", "descripiton2",
+                new BigDecimal("2.5"), 20, "2021-02-29T11:24:04.643", "2021-02-29T11:30:18.653");
+        certificate3 = new GiftCertificate(3L, "certificate3", "descripiton3",
+                new BigDecimal("3.5"), 30, "2021-01-29T11:24:04.643", "2021-01-29T11:30:18.653");
+    }
+
+    @Test
+    public void testGetAllSortedShouldReturnEntitySortedListWhenEntitiesArePresentInDb() {
```

Comment:
Your test names are very long. In that case, it's better to use underscore. One of the possible templates: test<methodName>_<condition>_<result>.
For example: testGetAll_entitiesArePresentInDb_entitySortedList

Reply:
commit 1518719